src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs

```diff
@@ -325,8 +326,11 @@ private void OnQueryButtonClick(object sender, RoutedEventArgs e)
 		}
 
 		private void SubmitSearch()
-		{
-			QuerySubmitted?.Invoke(this, new AutoSuggestBoxQuerySubmittedEventArgs(_suggestionsList.SelectedItem, Text));
+	    {	
+			string finalResult = GetObjectText(Text);
```

Comment:
I will create a new issue explaining the situation with SelectedItem for AutoSuggestBox and these PR changes can be the workaround for now i guess. Perhaps for now we should at least try to set it to SelectedItem and use the Text as a fallback? Something like: 

```suggestion
var finalResult = _suggestionsList.SelectedItem ?? GetObjectText(Text);
```